### PR TITLE
Clean up jobs code

### PR DESCRIPTION
**Pull request by rkondratowicz**
✈️ + no internet = this PR

See comments on diff below.

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sendletter/tasks/MarkLettersPostedTask.java [hunk 1]
```diff
@@ -32,11 +33,13 @@
...
-                                 IFtpAvailabilityChecker checker,
-                                 ReportParser parser,
-                                 AppInsights insights) {
+    public MarkLettersPostedTask(
+        LetterRepository repo,
+        FtpClient ftp,
+        IFtpAvailabilityChecker checker,
+        ReportParser parser,
+        AppInsights insights
+    ) {
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/sendletter/tasks/MarkLettersPostedTask.java]: to match formatting in the rest of the project code.

File: src/main/java/uk/gov/hmcts/reform/sendletter/tasks/MarkLettersPostedTask.java [hunk 2]
```diff
@@ -54,7 +57,7 @@ public void run(LocalTime now) {
                 .map(parser::parse)
                 .forEach(parsedReport -> {
                     insights.trackPrintReportReceived(parsedReport);
-                    parsedReport.statuses.forEach(this::updatePrintedAt);
+                    parsedReport.statuses.forEach(this::markAsPosted);
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/sendletter/tasks/MarkLettersPostedTask.java]: old producer would only set date, there was no 'status' column back then

File: src/main/java/uk/gov/hmcts/reform/sendletter/tasks/MarkLettersPostedTask.java [hunk 3]
```diff
@@ -70,25 +73,25 @@ public void run(LocalTime now) {
...
                 letter.setPrintedAt(Timestamp.from(letterPrintStatus.printedAt.toInstant()));
                 letter.setStatus(LetterStatus.Posted);
                 repo.save(letter);
-                logger.info("Marking letter {} as {}", letter.getId(), LetterStatus.Posted);
+                logger.info("Marked letter {} as posted", letter.getId());
             } else {
-                logger.info(
-                    "Skipping processing of letter {} with status {}",
+                logger.warn(
+                    "Failed to mark letter as posted {} - unexpected status: {}",
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/sendletter/tasks/MarkLettersPostedTask.java]: something weird happened if letter from report was not posted...

File: src/main/java/uk/gov/hmcts/reform/sendletter/tasks/MarkLettersPostedTask.java [hunk 4]
```diff
@@ -70,25 +73,25 @@ public void run(LocalTime now) {
...
+    private void markAsPosted(LetterPrintStatus letterPrintStatus) {
         Optional<Letter> optional = repo.findById(letterPrintStatus.id);
         if (optional.isPresent()) {
             Letter letter = optional.get();
             if (letter.getStatus() == LetterStatus.Uploaded) {
                 letter.setPrintedAt(Timestamp.from(letterPrintStatus.printedAt.toInstant()));
                 letter.setStatus(LetterStatus.Posted);
                 repo.save(letter);
-                logger.info("Marking letter {} as {}", letter.getId(), LetterStatus.Posted);
+                logger.info("Marked letter {} as posted", letter.getId());
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/sendletter/tasks/MarkLettersPostedTask.java]: should we not use enum? not too fuss about it

File: src/main/java/uk/gov/hmcts/reform/sendletter/tasks/MarkLettersPostedTask.java [hunk 5]
```diff
@@ -70,25 +73,25 @@ public void run(LocalTime now) {
...
+                logger.warn(
+                    "Failed to mark letter as posted {} - unexpected status: {}",
                     letter.getId(),
                     letter.getStatus()
                 );
             }
         } else {
             logger.warn(
-                "Failed to update printing date for letter {} - unknown letter",
+                "Failed to mark letter {} as posted - unknown letter",
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/sendletter/tasks/MarkLettersPostedTask.java]: same question here, applied rhetorically

File: src/main/java/uk/gov/hmcts/reform/sendletter/tasks/UploadLettersTask.java [hunk 6]
```diff
@@ -57,7 +57,11 @@ public void run() {
 
         do {
             letters = repo.findFirst10ByStatus(LetterStatus.Created);
-            letters.forEach(this::uploadLetter);
+            letters
+                .forEach(letter -> {
+                    uploadToFtp(letter);
+                    markAsUploaded(letter);
+                });
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/sendletter/tasks/UploadLettersTask.java]: 👏

File: src/main/java/uk/gov/hmcts/reform/sendletter/tasks/MarkLettersPostedTask.java (hunk 4 above)
Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/sendletter/tasks/MarkLettersPostedTask.java]: if we're not marking it as posted we should rename the class, method name and everything else :)

**Pull request merged**